docs/devsite-help/troubleshooting.md

```diff
@@ -218,3 +218,62 @@ native libraries are copied appropriately.
 
 This information is now in the [protocol buffers guide](protobuf.md#repeated-fields-and-maps); this section has been
 retained for the sake of existing links.
+
+## How can I get ErrorInfo from RpcException?
+
+`ErrorInfo` in `RpcException` provides a consistent and human readable error details.
+It contains three major fields - reason, domain and metadata described as below:
+
+- `Reason` - It is a constant value and indicates the proximate cause of 
+the error. Error reasons are unique within a particular domain of errors.
+
+- `Domain` - It is the logical grouping to which the error belongs. It is 
+typically the registered service name of the tool or product that 
+generates the error. Example: "pubsub.googleapis.com". If the error 
+is generated by some common infrastructure, the error domain must be 
+a globally unique value that identifies the infrastructure. 
+For Google API infrastructure, the error domain is "googleapis.com".
+
+- `Metadata`- It is the additional structured details about the error. 
+It can help in identifying the exact cause of the error.
+
+```csharp
+using Google.Api.Gax.Grpc;
+try
+{
+  // Code that can throw the RpcException.
+}
+catch (RpcException  ex)
+{
+   var errorInfo = ex.GetErrorInfo();   
+   var domain = errorInfo.Domain;
+   var reason = errorInfo.Reason;
+   var metadata = errorInfo.Metadata;
+
+   // Use the domain, reason and metadata of ErrorInfo, e.g. for logging or taking remedial action.
+}
+```
+
+### How can I get RpcException from SpannerException?
```

Comment:
Given the changes have been pushed recently and not released yet, we *may* want to release those changes and then say that "starting version 4.X.X or higher", so that users of existing versions of library understand they need to upgrade and this doesn't create confusion.

Reply:
Make sense. So should we wait for its release before merging this PR ?

Reply:
Yes, or split the PR into two: one just for RpcException and one for SpannerException.
Alternatively, given that this is completely Spanner-specific, we could move it into the Spanner docs - and merge that now, so that it'll be published in the next Spanner release. I think that would be better.

Reply:
Sure. I'll remove SpannerException section from troubleshoot.md and include it in Spanner docs in a separate PR once the help page with RpcException usage is published.